src/types.jl

```diff
@@ -69,8 +69,8 @@ ColorMixture{T}(Cs::NTuple{N,AbstractRGB}, channels::Vararg{Real,N}) where {T,N}
 @inline promote_typeof(obj1, obj2, objs...) = _promote_type(typeof(obj1), typeof(obj2), objs...)
 
 computeT(Cs::NTuple{N,AbstractRGB}, channels::NTuple{N,Real}) where {N} = eltype(promote_typeof(map(*, Cs, channels)...))
-ColorMixture(Cs::NTuple{N,AbstractRGB}, channels::NTuple{N,Real}) where {N} = ColorMixture{computeT(Cs, channels)}(Cs, channels)
-ColorMixture(Cs::NTuple{N,AbstractRGB}, channels::Vararg{Real,N}) where {N} = ColorMixture{computeT(Cs, channels)}(Cs, channels)
+Base.@constprop :aggressive ColorMixture(Cs::NTuple{N,AbstractRGB}, channels::NTuple{N,Real}) where {N} = ColorMixture{computeT(Cs, channels)}(Cs, channels)
```

Comment:
No, in this case the performance hit was about 100x. But that penalty was clearly dependent on specifics of the implementation since as stated in the previous comment it seems to have gone away.

The issue basically is that the RGB values are part of the type, and if we change the values in some way (e.g., `RGB{N0f8} -> RGB{Float32}`) then unless the *values* are aggressively const-propped then inference may not be able to calculate them. But now all operations return an element of the same type as the original, by default, so that seems to fix things.

Reply:
And the bigger story is that I'm now supporting 1.6. I think the caution is still important, though.